sql/query/expr/expr.go

```diff
@@ -119,3 +119,16 @@ type Operator interface {
 	SetRightHandExpr(Expr)
 	Token() scanner.Token
 }
+
+// Parentheses is a special expression which turns
+// any sub-expression as unary.
+// It hides the underlying operator, if any, from the parser
+// so that it doesn't get reordered by precedence.
```

Comment:
What does that mean exactly?

Reply:
In SQL, operators are ordered by precedence, for example, `*` has higher precedence than `+`, which means it must be evaluated first:

```python
3 + 4 * 2
# 4 * 2 is evaluated first
3 + 8
# then + is evaluated
11
```

If you want to avoid that, you can use parentheses

```python
(3 + 4) * 2
# 3 + 4 is evaluated first
7 * 2
# then * is evaluated
14
```

This behavior is done in the parser, when parsing the expression: https://github.com/genjidb/genji/blob/454cc983db979394c310a5a5df0e1180fb0f1447/sql/parser/expr.go#L30-L78

This function builds a tree based on operator precedence.
To avoid an expression in parentheses to have its precedence analyzed and taken into account by that function ([these two lines](https://github.com/genjidb/genji/blob/454cc983db979394c310a5a5df0e1180fb0f1447/sql/parser/expr.go#L70-L71)) we "hide" it by wrapping it in a type that **doesn't** implement the Operator interface so that in the eye of these two lines, that expression is unary, not binary.